### ISSUE TITLE
Try show an element whole if there is space

### ISSUE THREAD

**Issue by prototipo-dev** [QuestPDF/QuestPDF#1066]
Hello! First of all, thank you for the awesome library.

**Is your feature request related to a problem? Please describe.**
The issue is when there is an element of variable height that needs to be shown fully if possible

**Describe the solution you'd like**
Imagine when there is already a bit of text on a page and a table needs to be shown after it. This table does not fit in the remainder of the space the page has but would fit on a page by itself. I would like it to fit the element on a page by itself. If the element fits the current page, no change is needed, if the content does not fit the current page, try and fit it on the next page, if the content still does not fit, even on a page by itself, break the content to other pages (not throwing an exception).

**Describe alternatives you've considered**
I have tried using EnsureSpace, but since I don't know the space height that would be needed, and the header and footer might have dynamic heights, I could not make it work. 
I have also tried to use DynamicElement but the performance hit and the behavior was a bit inconsistent, this could need a more native approach.

**Additional context**
Basically the feature I am asking for is something like the KeepTogether used in some other PDF libraries, i.e https://kb.itextpdf.com/itext/how-to-prevent-splitting-a-table, but I could see this being useful for containers in general.

**Comment by MarcinZiabek** [comment]
Could you please try the `EnsureSpace` element? I am delighted to collaborate with you on learning more advanced requirements and adjusting existing implementations or adding new features.

**Comment by prototipo-dev** [comment]
Hello, I don't know how I could try to use the EnsureSpace element since I need to specify the height inside it, and I don't know the height that the table would occupy, the table can, in some cases, be taller than 1 whole page, but I don't know when that will happen. How would you advise me to implement the EnsureSpace in this case?

**Comment by prototipo-dev** [comment]
> Could you please try the `EnsureSpace` element? I am delighted to collaborate with you on learning more advanced requirements and adjusting existing implementations or adding new features.

Hello, I don't know how I could try to use the EnsureSpace element since I need to specify the height inside it, and I don't know the height that the table would occupy, the table can, in some cases, be taller than 1 whole page, but I don't know when that will happen. How would you advise me to implement the EnsureSpace in this case?

**Comment by MarcinZiabek** [comment]
The recent QuestPDF update includes significant improvements to the EnsureSpace element and introduces a new PreventPageBreak API. I believe these enhancements should address your requirement.

Please take a look at the updated documentation:
- [EnsureSpace API](https://www.questpdf.com/api-reference/ensure-space.html)
- [PreventPageBreak API](https://www.questpdf.com/api-reference/prevent-page-break.html)

Since there has been no recent activity, I’m closing this issue. If you have a similar request in the future, please feel free to open a new GitHub issue. Thanks for your feedback! ❤️
